presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analysis.java

```diff
@@ -614,22 +616,24 @@ public JoinUsingAnalysis getJoinUsing(Join node)
         return joinUsing.get(NodeRef.of(node));
     }
 
-    public void addTableColumnReferences(Identity identity, Multimap<QualifiedObjectName, String> tableColumnMap)
+    public void addTableColumnReferences(AccessControl accessControl, Identity identity, Multimap<QualifiedObjectName, String> tableColumnMap)
     {
-        Map<QualifiedObjectName, Set<String>> references = tableColumnReferences.putIfAbsent(identity, new LinkedHashMap<>());
+        AccessControlInfo accessControlInfo = new AccessControlInfo(accessControl, identity);
+        Map<QualifiedObjectName, Set<String>> references = tableColumnReferences.computeIfAbsent(accessControlInfo, (k) -> new LinkedHashMap<>());
```

Comment:
can't use LinkedHashMap::new because it can't resolve the constructor (the function takes an argument that we don't used, and there are single argument constructors for LinkedHashMap)